### ISSUE TITLE
"Unable to find a backend server" in the default server group

### ISSUE THREAD

**Issue by krizhanovsky** [tempesta-tech/tempesta#1650]
# Motivation

https://github.com/tempesta-tech/tempesta/wiki/Backend-servers makes example for default server group. Meantime the test https://github.com/tempesta-tech/tempesta-test/pull/230#discussion_r896941645 exhibits that the logic is broken.

# Scope

```bash
root@ubuntu:~/tempesta# git log |head -1
commit cda8ef83b803bf0ac5d93ec87947d45595c7a98c
root@ubuntu:~/tempesta# cat etc/tempesta_fw.conf 
server 192.168.100.4:8000 conns_n=4;
```

If Tempesta FW is compiled with `DEBUG=1`, then dmesg shows `Unable to find a backend server` on a browser request. 

`Unable to find a backend server` is printed for `tfw_http_get_srv_conn()` returning `NULL`. We don't have a session, so this seems means that `tfw_vhost_get_srv_conn()` returns `NULL`. On `DEBUG=3` I didn't see `vhost: use server group` message from the last function, so the problem is in empty `main_sg` server group. IIUC for the above config `tfw_vhost_cfgend()` should have set the default server group.

# Testing

Test modified in https://github.com/tempesta-tech/tempesta-test/pull/230 must work without the config modifications

# Documentation

Need to carefully analyze what's going on with the configuration process and request handling, probably we should adjust our documentation instead of changing the code. Anyway, let's discuss what's going on before changing the code.

**Comment by const-t** [comment]
In case of analyzing configuration process we should take a look at `vhost *`.

With config:
```
listen 443 proto=h2;

srv_group default {
    server ${general_ip}:8000;
}

vhost default {
    tls_match_any_server_name;
    tls_certificate ${tempesta_workdir}/tempesta.crt;
    tls_certificate_key ${tempesta_workdir}/tempesta.key;

    proxy_pass default;
}

http_chain {
-> default;
}
```
We get this kind of error:
```
ERROR: tls_match_any_server_name: directive can be applied only to '*' vhost.
ERROR: configuration parsing error:
        8:     tls_match_any_server_name;
```
As I understand, `vhost *` is same as `vhost default`. This should be clarified.

**Comment by pale-emperor** [comment]
I trying to configure tempesta to forward websockets connections to 8765 port and that what i figured out:

1. Tempesta can start with `completely empty config`
RESULT: port (80) is listening by default but every connection provide dmesg Warning (Maybe its okay, cause i not specified server for connections)
DMESG: `Warning: request dropped: processing error, status 502: 127.0.0.1`

2. Trying to run tempesta with only server directive
`server 127.0.0.1:8765 conns_n=4;`
RESULT: Listen 80 port but still dmesg Warning on every connection - connections **not forwarding** to specified 127.0.0.1:8765
DMESG: `Warning: request dropped: processing error, status 502: 127.0.0.1`

3. Minimal working configuration looks:
```
server 127.0.0.1:8765 conns_n=4; - default srv_group succesfullly created

vhost default {
    proxy_pass default; - specify srv_group
}

http_chain {
    -> default; - specify vhost
}
```

RESULT: Thats only correct way that i can connect WS

4. Same configuration (as 3) without `http_chains `
```
server 127.0.0.1:8765 conns_n=4; - default srv_group succesfullly created
vhost default {
    proxy_pass default;
}
```
Make every connection provide dmesg Warning
DMESG: `Warning: request dropped: processing error, status 502: 127.0.0.1
`

Conclusion

Every directive seems like optional, but only way to make connections is (Example 3):
`http_chain -> vhost -> srv_group`

http_chain - not created by default, not working without vhost
vhost - not created by default
srv_group - created by default if server directive is present

**Comment by nickzaev** [comment]
@pale-emperor I've blantly assumed that autocreating both default `vhost` and `http_chain` would actually fix a bunch of tests on its own, what do you think, is that the case? If doesn't take that much of effort to try, should we test it?

**Comment by pale-emperor** [comment]
@nickzaev  I don't think we should make vhost and http_chain creation automatically just for the sake of fixing tests. It's just not entirely clear right now, whether http_chain and vhost are required to work properly and in that connection  - i have no idea how to fix test properly.

If we come to an agreement which config is considered the minimum for normal operation - based on this, I will try to fix the tests

Here is just information about which config leads to the desired result. I'm not leading here to any specific solution.

**Comment by krizhanovsky** [comment]
Probably that's not much sense for us to run with empty config. For example, in this case we have to suppose that there is some server running e.g. on `127.0.0.1:8000`, which might confuse users even more if they don't have the implicit server.
